statsmodels/sandbox/mice/mice.py

```diff
@@ -0,0 +1,1295 @@
+        The number of imputed datasets to skip between consecutive
+        imputed datasets that are used for analysis.
+    init_kwds : dict-like
+        Dictionary of keyword arguments passed to the init method
+        of the analysis model.
+    fit_kwds : dict-like
+        Dictionary of keyword arguments passed to the fit method
+        of the analysis model.
+
+    Examples
+    --------
+    Simple example using defaults:
+    %(mice_example_1)s
+
+    Example using an iterator:
+    %(mice_example_2)s
+    """ % {'mice_example_1' : _mice_example_1,
+           'mice_example_2' : _mice_example_2}
+
+
+    def __init__(self, model_formula, model_class, data, n_skip=3,
+                 init_kwds=None, fit_kwds=None):
+
+        self.model_formula = model_formula
+        self.model_class = model_class
+        self.n_skip = n_skip
+        self.data = data
+        self.results_list = []
+
+        self.init_kwds = init_kwds if init_kwds is not None else {}
+        self.fit_kwds = fit_kwds if fit_kwds is not None else {}
+
+
+    def __iter__(self):
+        return self
+
+
+    def next(self):
+        return self.__next__()
+
+
+    def __next__(self):
+
+        self.data.update_all(self.n_skip)
+        start_params = None
+        if len(self.results_list) > 0:
+            start_params = self.results_list[-1].params
+        model = self.model_class.from_formula(self.model_formula,
+                                              self.data.data,
+                                              **self.init_kwds)
+        self.fit_kwds.update({"start_params": start_params})
+        result = model.fit(**self.fit_kwds)
+
+        return result
+
+
+    def fit(self, n_burnin=10, n_imputations=10):
+        """
+        Fit a model using MICE.
+
+        Parameters
+        ----------
+        n_burnin : int
+            The number of burn-in cycles to skip.
+        n_imputations : int
+            The number of data sets to impute
+        """
+
+        # Run without fitting the analysis model
+        self.data.update_all(n_burnin)
+
+        for j in range(n_imputations):
+            result = next(self)
+            self.results_list.append(result)
+
+        self.endog_names = result.model.endog_names
+        self.exog_names = result.model.exog_names
+
+        return self.combine()
+
+
+    def combine(self):
+        """
+        Pools MICE imputation results.
+
+        This method can only be used after the `run` method has been
+        called.  Returns estimates and standard errors of the analysis
+        model parameters.
+
+        Returns a MICEResults instance.
+        """
+
+        # Extract a few things from the models that were fit to
+        # imputed data sets.
+        params_list = []
+        cov_list = []
+        scale_list = []
+        for results in self.results_list:
+            params_list.append(np.asarray(results.params))
+            cov_list.append(np.asarray(results.cov_params()))
```

Comment:
We could save here the asarray detour if we use results._results, which is the results without wrapper.

(In cases where we could have wrapped or unwrapped results, we check it `results = getattr(results._results, results` might work)